### PR TITLE
Add documentation for Eventing Authorization

### LINKED ISSUE #6099: Add documentation for Authorization support in Knative-Eventing
## What is changing?

Knative Eventing will support Authorization via EventPolicies.
The feature track is available [here](https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?).

## How will this impact our users?

Users can define EventPolicies to restrict who can send events to a (knative-eventing owned) Sink.

## In what release will this take happen (to the best of your knowledge)?
1.16

## Context
Project: https://github.com/orgs/knative/projects/79/views/1

### PR DESCRIPTION
**Pull request by creydr**
Fixes #6099

## Proposed Changes

- :gift: Add documentation for [Eventing Authorization feature](https://github.com/orgs/knative/projects/79/views/1)

### CODE REVIEW

**Comment by creydr** [comment]
/cc @pierDipi @matzew 
Could you give this an initial review?
-> https://deploy-preview-6104--knative.netlify.app/docs/eventing/features/authorization/

**Comment by creydr** [comment]
We probably want to wait for
* https://github.com/knative-extensions/eventing-kafka-broker/issues/4038
* https://github.com/knative-extensions/eventing-kafka-broker/issues/4039
before merging this, so that we have the full support in Eventing-Kafka-Broker 

/hold

File: docs/eventing/features/authorization.md
Comment (pierDipi) [inline docs/eventing/features/authorization.md:94]: `to` Trigger doesn't make sense?

File: docs/eventing/features/authorization.md
Comment (pierDipi) [inline docs/eventing/features/authorization.md:126]: let's show an example with filters? 

I'd also describe what happens with multiple policies combined with filters

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:94]: Uh. Absolutely. Will update it

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:126]: In line 59 we have an example with filters. Or do you mean in the later "bigger" example?

File: docs/eventing/features/authorization.md
Comment (Cali0707) [inline docs/eventing/features/authorization.md:12]: Doesn't this also require `transport-encyption` to be `strict`?

Comment (creydr) [reply docs/eventing/features/authorization.md:12]: AuthZ not directly. Only AuthN is not secure without it (that's described on the sender-identity feature page)

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:126]: > I'd also describe what happens with multiple policies combined with filters
Can you describe what you mean? Shouldn't it be the same on multiple policies with or without filters? As long as one policy allows the request (having filters or not), the request is passed

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:12]: But I can add a not there too

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md]: Updated

**Comment by Cali0707** [comment]
I think for the filters, it's normally "All" of the filters that just be true for a specific policy, not one of the filters for a policy

File: docs/eventing/features/authorization.md
Comment (lberk) [inline docs/eventing/features/authorization.md:78]: Is the `to.*` fields logically combined in an `or` statement? or an `and` statement?  I'm not sure from the description if specifying multiple `to.*` targets widens the scope of the event policy, or narrows it.

File: docs/eventing/features/authorization.md
Comment (lberk) [inline docs/eventing/features/authorization.md:126]: Is there any other subject other than a service account?

File: docs/eventing/features/authorization.md
Comment (lberk) [inline docs/eventing/features/authorization.md:237]: `namespace-2` ?

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:78]: Interesting point of view to this. Let me try to make it clearer. 
For background information: AuthZ is checked on the receiving side, so we don't know at this point, where the events got sent to elsewhere.

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:126]: Not yet. But when we integrate this with other IDPs and allow sending events from outside the mesh, this would be more relevant.
But to not confuse, I'd better remove it

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md:237]: :+1:

Comment (creydr) [reply docs/eventing/features/authorization.md:237]: Depending on the step in the example, events from this source will be allowed or rejected. So I tried to make it clearer in the different steps. I also tried to make it clear in the image, with the "architecture overview" (not sure, if you checked the code or the rendered output of this PR :smile:  )

File: docs/eventing/features/authorization.md
Comment (creydr) [reply docs/eventing/features/authorization.md]: Updated. Does this make it clearer?

**Comment by creydr** [comment]
/unhold
as we enabled the support in EKB

File: docs/eventing/features/authorization.md
Comment (Cali0707) [inline docs/eventing/features/authorization.md:206]: I think it's currently implied here, but maybe we could make it clear that when multiple policies apply to the same resource, if the event matches one of the policies the event will be delivered?

Comment (creydr) [reply docs/eventing/features/authorization.md]: Good point to make it clearer :+1: Updated it in 993c2a39a6226ca92122df6dfdf9e02f30d4df99

**Review by Cali0707 — APPROVED** [review]
/lgtm
/approve
/hold for other reviews

**Comment by creydr** [comment]
@pierDipi @lberk any other comments, or can we unhold this?

**Comment by pierDipi** [comment]
/lgtm

Thanks!

**Pull request merged**
